Microsoft.Toolkit.Uwp.UI.Media/Geometry/Parsers/ColorParser.cs

```diff
@@ -0,0 +1,140 @@
+        /// </summary>
+        /// <param name="colorString">Color string in Hexadecimal or HDR format</param>
+        /// <returns>Color</returns>
+        internal static Color Parse(string colorString)
+        {
+            var match = RegexFactory.ColorRegex.Match(colorString);
+            if (!match.Success)
+            {
+                throw new ArgumentException("Invalid Hexadecimal string!", nameof(colorString));
+            }
+
+            return Parse(match);
+        }
+
+        /// <summary>
+        /// Attempts to convert color string in Hexadecimal or HDR color format to the
+        /// corresponding Color object.
+        /// The hexadecimal color string should be in #RRGGBB or #AARRGGBB format.
+        /// The '#' character is optional.
+        /// The HDR color string should be in R G B A format.
+        /// (R, G, B &amp; A should have value in the range between 0 and 1, inclusive)
+        /// </summary>
+        /// <param name="colorString">Color string in Hexadecimal or HDR format</param>
+        /// <param name="color">Output Color object</param>
+        /// <returns>True if successful, otherwise False</returns>
+        internal static bool TryParse(string colorString, out Color color)
+        {
+            var match = RegexFactory.ColorRegex.Match(colorString);
+            if (!match.Success)
+            {
+                return false;
+            }
+
+            color = Parse(match);
+            return true;
+        }
+
+        /// <summary>
+        /// Converts a Vector4 High Dynamic Range Color
+        /// to Color. Negative components of the Vector4 will be
+        /// sanitized by taking the absolute value of the component.
+        /// The HDR Color components should have value in the range
+        /// between 0 and 1, inclusive. If they are more than 1, they
+        /// will be clamped at 1.
+        /// </summary>
+        /// <param name="hdrColor">High Dynamic Range Color</param>
+        /// <returns>Color</returns>
+        internal static Color Parse(Vector4 hdrColor)
+        {
+            // Vector4's X, Y, Z, W components match to
+            // Color's   R, G, B, A components respectively
+            return Parse(hdrColor.X, hdrColor.Y, hdrColor.Z, hdrColor.W);
+        }
+
+        /// <summary>
+        /// Converts the given HDR color values to Color.
+        /// </summary>
+        /// <param name="x">Red Component</param>
+        /// <param name="y">Green Component</param>
+        /// <param name="z">Blue Component</param>
+        /// <param name="w">Alpha Component</param>
+        /// <returns>Instance of Color.</returns>
+        internal static Color Parse(float x, float y, float z, float w)
+        {
+            var r = (byte)Math.Min(Math.Abs(x) * 255f, 255f);
+            var g = (byte)Math.Min(Math.Abs(y) * 255f, 255f);
+            var b = (byte)Math.Min(Math.Abs(z) * 255f, 255f);
+            var a = (byte)Math.Min(Math.Abs(w) * 255f, 255f);
```

Comment:
This can be vectorized to result in much better codegen 🚀
See JIT diff on .NET 5 [here](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgDkBXfGKASzFwBuGvSasAShwB2GXtxYBhCPgAOvADY8AyjwBu/GEJoiAzE1IMFNAN40G9hrxk8p2dWKSOZDAArYouDDkABQAZuoQ2BgMCGgM4ZHRAJ5xCVEMAF6pEekA7gCUdg621A5lDLr+DFAMALwMwcBJGDD5ALJRABYsbU7BHRjdAILAuMEI+QwAVAykAKxzoXHzi/nCpeX2lTUA5nUNTS3tXT19A8OjwUmTMytLswuha0Wb2wzA+43NreenUv0nEZjDI3B6LZaPZ4bV5VbCfQ4/E69f6/IHBArTMH3O5Ql7lYgAdgYADEIBBgtg4lA4js4sAoWUAL54vFOFpQVzuZieNm+fyBUhhHLRWLxYUMFJixKZbLSgp46ybex4soANRgYAw0BQFR19SkMFy4zikqyDAK6yVDBVDl0eoY6s12r+wUdWqgKBYaLtoLucQNRpxDKVNq2VRq9S+RztLAAGpalW89pGEfkYwBNBMwmofFPfNOegBaWfKbzheejnoA6utQ0wiaTyZTqjS6cH7MzodauwBtNowQYQAAmAElVOp+gPOsOxyp1AB5FSyCBSXDsCAjqTqJxOHb5AC6eOIZm5XmijajMAYzYR7zp3wYYHvLQYQ8KXZKVsJDDoJc7nZoIA=) (works on .NET Nativee too).
```suggestion
            Vector4 v4 = new(x, y, z, w);

            v4 = Vector4.Min(Vector4.Abs(v4) * 255f, new(255f));

            var r = (byte)v4.X;
            var g = (byte)v4.Y;
            var b = (byte)v4.Z;
            var a = (byte)v4.W;
```

Reply:
Updated.